h2o-core/src/test/java/water/util/FileUtilsTest.java

```diff
@@ -0,0 +1,68 @@
+package water.util;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.runner.RunWith;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.runner.CloudSize;
+import water.runner.H2ORunner;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+@RunWith(H2ORunner.class)
+@CloudSize(1)
+public class FileUtilsTest {
+
+    @Rule
+    public EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
+    @Test
+    public void testFindFileInPredefinedPath() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
```

Comment:
the method has quite a few branches and a regex I would like to see some more unit tests

Reply:
Added. Please have a look now.

Reply:
Testing both positive (file found) and negative scenarios now. Both with "/" at the end of the directory-of-search path and without it.